packages/migration/src/migrations/v0002_dataPixelTo1080.ts

```diff
@@ -22,41 +28,51 @@ const OLD_PAGE_HEIGHT = 732;
 const SCALE_X = NEW_PAGE_WIDTH / OLD_PAGE_WIDTH;
 const SCALE_Y = NEW_PAGE_HEIGHT / OLD_PAGE_HEIGHT;
 
-function dataPixelTo1080({ pages, ...rest }) {
+export type StoryV2 = StoryV1;
+export type PageV2 = PageV0;
+export type UnionElementV2 = UnionElementV0;
+
+function dataPixelTo1080({ pages, ...rest }: StoryV1): StoryV2 {
   return {
     pages: pages.map(reducePage),
     ...rest,
   };
 }
 
-function reducePage({ elements, ...rest }) {
+function reducePage({ elements, ...rest }: PageV0): PageV0 {
   return {
     elements: elements.map(updateElement),
     ...rest,
   };
 }
 
-function updateElement({ x, y, width, height, fontSize, ...rest }) {
+function updateElement({
+  x,
+  y,
+  width,
+  height,
+  ...rest
+}: UnionElementV0): UnionElementV0 {
   const element = {
     x: dataPixels(x * SCALE_X),
     y: dataPixels(y * SCALE_Y),
     width: dataPixels(width * SCALE_X),
     height: dataPixels(height * SCALE_Y),
     ...rest,
   };
-  if (typeof fontSize === 'number') {
-    element.fontSize = dataPixels(fontSize * SCALE_Y);
+  if ('fontSize' in element && typeof element.fontSize === 'number') {
```

Comment:
I feel like this should be done with type guard instead:

```ts
function isTextElement(e:ElementV1):e is TextElementV1 {
  return 'font' in e;
}
...
if (isTextElement(element) && typeof element.fontSize === 'number') {
```

This would also mean that the union type isn't required at all. I actually really want to avoid that union type. I feel like the types here shouldn't be exhaustive but minimalistic - so interfaces over types. That does require these guards all over though, but that's also kind of required for narrowing a union type.

Reply:
Hm, but it's not correct like this though? Since the function actually can take all types, not just the text element.

What happens if there's a prop that's being used in multiple element types, like `resource`? Or `flip`? Also, the returned element can still be any element type, this would mean a lot of casting everywhere.

Reply:
Also, I understand that generally we're using just the `Element` type for `Page` since potentially there might be other elements added so better to not use the union type, however, in these migration files we had those specific element types so it seems incorrect to use a generic base `Element`. Not sure what would be the benefit of using a minimalistic type in this case if these migration files were created for very specific element types. Thoughts?

Reply:
Note: we discussed this in the backlog grooming and decided to convert a few files using the type guarding to see how it seems.

@barklund PTAL

Reply:
Conclusion: still going to use the union type as before since otherwise TS won't really recognize nor accurately alert about types not matching and we'd have to do a lot of casting for getting rid of the TS errors.